File: exercises/java/src/main/java/cards/Deck.java
package cards;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public abstract class Deck {
    protected List<Card> cards = new ArrayList<>();

    public void shuffle() {
        Collections.shuffle(cards);
    }

    public String[] getCards() {
        String[] result = new String[cards.size()];
        int cardNumber = 0;
        for (Card card : cards) {
            result[cardNumber] = card.toString();
            cardNumber++;
        }
        return result;
    }

    public Card deal() {
        return cards.remove(0);
    }
}

File: exercises/dotnet/McrDigital.Bootcamp1.Cards/Deck.cs
using System;
using System.Collections.Generic;

namespace McrDigital.Bootcamp1.Cards
{
    public abstract class Deck
    {
        protected List<Card> _cards;

        protected Deck()
        {
            _cards = new List<Card>();
        }

        public string[] GetCards()
        {
            var result = new String[_cards.Count];


            var cardNumber = 0;
            foreach (var card in _cards)
            {
                result[cardNumber] = card.ToString();
                cardNumber++;
            }

            return result;
        }

        public Card Deal()
        {
            var card = _cards[0];
            _cards.RemoveAt(0);
            return card;
        }

        public void Shuffle()
        {
            _cards.KnuthShuffle();
        }
    }
}